src/tests/network-api-test/network-api-test.c

```diff
@@ -0,0 +1,603 @@
+    actual = OS_SocketAddrGetPort(&PortNum, NULL);
+    UtAssert_True(actual == expected, "OS_SocketAddrGetPort() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    /* OS_SocketGetIdByName */
+    expected = OS_INVALID_POINTER; 
+    actual = OS_SocketGetIdByName(NULL,NULL);
+    UtAssert_True(actual == expected, "OS_SocketGetIdByName() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_NAME_NOT_FOUND; 
+    actual = OS_SocketGetIdByName(&objid,"NF");
+    UtAssert_True(actual == expected, "OS_SocketGetIdByName() (%ld) == OS_ERR_NAME_NOT_FOUND", (long)actual);
+
+    /* OS_SocketGetInfo */
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketGetInfo(1, NULL);
+    UtAssert_True(actual == expected, "OS_SocketGetInfo() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_INVALID_ID;
+    actual = OS_SocketGetInfo(0, &prop);
+    UtAssert_True(actual == expected, "OS_SocketGetInfo() (%ld) == OS_ERR_INVALID_ID", (long)actual);
+
+} /* end TestDatagramNetworkApi */
+
+/*****************************************************************************
+ *
+ * Datagram Network Teardown
+ *
+ *****************************************************************************/
+void TestDatagramNetworkApi_Teardown(void)
+{
+    /* Close sockets */
+    OS_close(p1_socket_id);
+    OS_close(p2_socket_id);
+
+} /* end TestDatagramNetworkApi_Teardown */
+
+
+/*****************************************************************************
+ *
+ * Server Function for Stream Network Functional Test
+ *
+ *****************************************************************************/
+void Server_Fn(void)
+{
+    OS_SockAddr_t addr;
+    char Buf_send_s[4] = {0};
+    char Buf_rcv_s[4] = {0};
+    char Buf_trans[8] ={0};
+    uint32 connsock_id = 0;
+
+    /* Accept incoming connections */
+    OS_SocketAccept(s_socket_id, &connsock_id, &addr, OS_PEND);
+
+    /* Once connected, send data to client */
+    strcpy(Buf_send_s, "abc");
+    OS_TimedWrite(connsock_id, Buf_send_s, sizeof(Buf_send_s), 10);
+
+   /* Recieve incoming data from client*/
+    OS_TimedRead(connsock_id, Buf_rcv_s, sizeof(Buf_rcv_s), 10);
+
+   /* Transform the incoming data and send it back to client */
+    strcpy(Buf_trans, "uvw");
+    strcat(Buf_trans, Buf_rcv_s);
+    OS_TimedWrite(connsock_id, Buf_trans, sizeof(Buf_trans), 10);
+
+
+    OS_close(s_socket_id);
+
+} /* end Server_Fn */
+
+
+/*****************************************************************************
+ *
+ * Stream Network Functional Test 
+ *
+ *****************************************************************************/
+void TestStreamNetworkApi(void)
+{
+    int32 status; 
+    int32 expected;
+    int32 actual;
+    uint32 loopcnt;
+    uint32 temp_id;
+    OS_SockAddr_t temp_addr;
+    OS_task_prop_t taskprop;
+    char Buf_rcv_c[4] = {0};
+    char Buf_send_c[4] = {0};
+    char Buf_expected[4] ={0};
+    char Buf_rcv_trans[8] = {0};
+    char Buf_expec_trans[8] ={0};
```

Comment:
Did you see my previous comment about passing non-strings to ensure that the interface is 8-bit clean?  Recommendation is to pass some non-alphanumeric/binary chars too, to ensure that embedded NUL chars are passed, and no CRLF translation is done, etc.

Just make these buffers into length of 256 and pass every possible char value.